deposit-transactions.md

```diff
@@ -0,0 +1,165 @@
+   function with correct values associated with the corresponding L1 block.
+
+## Special Accounts on L2
+
+The L1 Attributes Deposit Transaction involves two special purpose accounts:
+
+1. The L1 Attributes Depositor Account
+2. The L1 Attributes Predeploy
+
+### L1 Attributes Depositor Account
+
+[l1-attributes-depositor-account]: #l1-attributes-depositor-account
+
+The Depositor Account is an EOA with no known private key. It has the address
+`0xdeaddeaddeaddeaddeaddeaddeaddeaddead0001`. Its value returned by the `CALLER` and `ORIGIN`
+opcodes during execution of the L1 Attributes Deposit Transaction.
+
+### L1 Attributes Predeploy
+
+[l1-attributes-predeploy]: #l1-attributes-predeploy
+
+A predeployed contract on L2 at address `0x4200000000000000000000000000000000000014`, which holds
+certain block variables from the corresponding L1 block in storage, so that they may be accessed
+during the execution of the subsequent deposited transactions.
+
+The contract implements an authorization scheme, such that it only accepts state-changing calls from
+the [Depositor Account].
+
+The contract has the following solidity interface, and can be interacted with according to the
+[contract ABI specification][ABI].
+
+[ABI]: https://docs.soliditylang.org/en/v0.8.10/abi-spec.html
+
+```solidity
+interface L1BlockValues {
+
+  function setL1BlockValues(
+    uint256 number,
+    uint256 timestamp,
+    uint256 baseFee,
+    bytes32 hash
+  ) external;
+
+  function l1Number() view;
+  function l1Timestamp() view;
+  function l1BaseFee() view;
+  function l1Hash() view;
+}
+```
+
+## Deposited Transactions
+
+[deposited-transactions]: #deposited-transactions
+
+"Deposited Transactions" are derived from logs emitted by the [Deposit Feed
+contract][deposit-feed-contract] on L1.
+
+### Deposit Feed Contract
+
+[deposit-feed-contract]: #deposit-feed-contract
+
+The Deposit Feed contract is deployed to L1. Deposited Transactions are derived from the values in
+the `TransactionDeposited` event(s) emitted by the Deposit Feed contract.
+
+The Deposit Feed handles two special cases:
+
+1. A `to` value of `0`, which results in a contract creation on L2.
+2. A call from a contract account, in which case the `from` value is transformed to its L2 alias.
+   This prevents attacks in which a contract on L1 has the same address as a contract on L2 but
+   doesn't have the same code. We can safely ignore this for EOAs because they're guaranteed to have
+   the same "code" (i.e. no code at all). This also makes it possible for users to interact with
+   contracts on L2 even when the Sequencer is down.
+
+A solidity like pseudocode implementation demonstrates the functionality:
+
+```solidity
+contract DepositFeed {
+  event TransactionDeposited(
+    address indexed from,
+    address indexed to,
+    uint256 value,
+    bytes _data
+  );
+
+  event TransactionDeposited(
+    address indexed from,
+    uint256 value,
+    bytes _data
+  );
```

Comment:
I created two events because felt it was important to differentiate between `address(0)` and `null`. However on reflection, the impl below makes it impossible to emit the first event with `to == address(0)`, so it doesn't really matter. 
Questions: 
Should it be possible to send to `address(0)` on L2? 
Should we consider adding an `isCreation` boolean argument?

Reply:
> Should it be possible to send to address(0) on L2?

Yes, although it's weird. Adding a byte for bitflags like `isCreation` and possible future toggles does not sound so bad to me.

Reply:
Seems quite important to allow `address(0)` since it's used for burning tokens a lot. No reason that shouldn't be allowed for deposited transactions.

I think having a single event with the boolean flag will be simpler to handle in the rollup node, no?

Reply:
We currently do not support L1 -> L2 contract creations. In the case of the sequencer going down, it is nice functionality to have. Is it worth the extra complexity from a user perspective when a contract factory on L2 could be used to do a L1 -> L2 deployment?

Reply:
It's a good question.
- Option A: add an extra field in the even and an extra argument in the function (unless there's a way to detect the nul address from the argument?).  We'll alsoneed to check the field in the new transaction type handler and call contract creation logic. And finally, increment nonces on the basis of deposits too.
- Option B: Users have to call a predeployed contract factory instead.

Option B seems simpler, though a bit less identical to what you'd do on L1. I could be convinced either way.

Reply:
Reposting another relevant comment made [below](https://github.com/ethereum-optimism/optimistic-specs/pull/54/files#r765064416) by @ben-chain:
> I think using addr(0) is probably fine, but raising this issue we've seen in the past. I don't think it's the same issue, but there's always risk of confusion where a dev allows an L1 contract to call an arbitrary L2 contract, and a user can trick the contract into CREATEing instead of making a call to addr(0)